site/fs.js

```diff
@@ -0,0 +1,49 @@
+const fs = require('fs').promises
+const path = require('path')
+const rimraf = require('rimraf')
+const util = require('util')
+
+const rimrafAsync = util.promisify(rimraf)
+
+const copyDirRecursiveAsync = async (src, dest) => {
+  try {
+    fs.mkdir(dest, { recursive: true })
+  } catch (_) {
+    // ignore erros for mkdir
+  }
+
+  const childrenItems = await fs.readdir(src)
```

Comment:
This has the potential to create a lot of promises.
I don't think we should optimise this yet since it's only used for our docs site that doesn't contain many of files.